src/io_worker.cpp

```diff
@@ -281,13 +282,17 @@ void IOWorker::on_execute(uv_async_t* async) {
 #endif
   IOWorker* io_worker = static_cast<IOWorker*>(async->data);
 
-  RequestHandler* request_handler = NULL;
+  RequestHandler* temp = NULL;
   size_t remaining = io_worker->config().max_requests_per_flush();
-  while (remaining != 0 && io_worker->request_queue_.dequeue(request_handler)) {
-    if (request_handler != NULL) {
+  while (remaining != 0 && io_worker->request_queue_.dequeue(temp)) {
+    RequestHandler::Ptr request_handler(temp);
```

Comment:
See above for why the queue uses `RequestHandler*` instead of `RequestHandler::Ptr`. 

The goal was to get the `RequestHandler` into a shared ptr as soon as possible to avoid error (in future changes). Also, the `SepeculativeExecution`constructor takes a shared ptr.